docs/pytree_arguments.rst

```diff
@@ -0,0 +1,67 @@
+Pytree arguments to JAX functions
+=================================
+
+What is a pytree?
+^^^^^^^^^^^^^^^^^
+
+In JAX, a pytree is **a container of leaf elements and/or more pytrees**.
+Containers include lists, tuples, and dicts (JAX can be extended to consider
+other container types as pytrees, see the "JAX pytrees" notebook under "Advanced
+JAX Tutorials"). A leaf element is anything that's not a pytree, e.g. an
+array. In other words, a pytree is just **a possibly-nested standard Python
+container**.  If nested, note that the container types do not need to match. A
+single "leaf", i.e. a non-container object, is also considered a pytree.
+
+Example pytrees::
+
+  [1, "a", object()] # 3 leaves
+
+  (1, (2, 3), ()) # 3 leaves
+
+  [1, {"k1": 2, "k2": (3, 4)}, 5] # 5 leaves
+
+Pytrees and JAX functions
+^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Many JAX functions, including all function transformations, operate over pytrees
+of arrays (other leaf types are sometimes allowed as well). Transformations are
+only applied to the leaf arrays while preserving the original pytree structure;
+for example, ``vmap`` and ``pmap`` only map over arrays, but automatically map
+over arrays inside of standard Python sequences, and can return mapped Python
+sequences.
+
+Applying optional parameters to pytrees
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Some JAX function transformations take optional parameters that specify how
+certain input or output values should be treated (e.g. the ``in_axes`` and
+``out_axes`` arguments to ``vmap``). These parameters are also pytrees, and the
+leaf values are "matched up" with the corresponding input and output leaf
```

Comment:
Technically it is "input and output leaf or subtree", but maybe that it too much at this point.